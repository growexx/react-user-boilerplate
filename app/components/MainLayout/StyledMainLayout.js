import styled from 'styled-components';
export const StyledMainLayout = styled.div`
  .ant-layout-sider {
    transition: none;
    background: #190426;
  }
  .ant-menu .ant-menu-item,
  .ant-menu-submenu-title,
  .ant-menu {
    transition: none;
  }
  .ant-menu.ant-menu-dark,
  .ant-menu-dark .ant-menu-sub,
  .ant-menu.ant-menu-dark .ant-menu-sub {
    background-color: #190426;
    color: @secondary-color;
  }
  .ant-menu-dark.ant-menu-dark:not(.ant-menu-horizontal)
    .ant-menu-item-selected {
    background-color: @primary-color;
    color: @white;
  }
  .ant-menu-dark .ant-menu-item,
  .ant-menu-dark .ant-menu-item-group-title,
  .ant-menu-dark .ant-menu-item > a,
  .ant-menu-dark .ant-menu-item > span > a {
    color: @white;
  }
  .ant-menu-item-selected > span > a {
    color: @white !important;
  }
  .sideBarTrigger {
    padding: 0 @padding-lg;
    font-size: @font-size-base+4px;
    line-height: 64px;
    cursor: pointer;
    transition: color 0.3s;
    outline: none;
  }
  .sideBarTrigger:hover {
    color: @primary-color;
  }
  #components-layout-demo-custom-trigger .logo {
    display: flex;
    justify-content: center;
    align-items: center;
    img {
      margin: @margin-md;
    }
  }
  .site-layout .site-layout-background {
    background: #fff;
  }
  .headerLayout {
    display: flex;
    padding: 0;
    background-color: @white;
  }
  &[data-environment] {
    &:before {
      content: attr(data-environment);
      background: #ff8c1e;
      height: 25px;
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
      z-index: 200;
      color: #fff;
      text-align: center;
      text-transform: uppercase;
    }

    > .ant-layout {
      position: relative;
      top: 25px;
    }
  }

  &[data-environment='development'] {
    &:before {
      background: #b6c5cd;
    }
  }
`;

export const ToggleBreadCrumb = styled.div`
  display: flex;
`;
