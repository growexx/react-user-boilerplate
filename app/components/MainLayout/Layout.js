/**
 * Layout.js
 *
 */

import React from 'react';
import { Layout, Menu } from 'antd';
import { MenuUnfoldOutlined, MenuFoldOutlined } from '@ant-design/icons';
import PropTypes from 'prop-types';
import App from 'containers/App';
import Footer from 'components/Footer';
import AppHeader from 'components/Header';
import SideBar from 'components/SideBar';
import { getUserData } from 'utils/Helper';
import { ROUTES } from 'containers/constants';
import { Link, withRouter } from 'react-router-dom';
import styled from 'styled-components';
import { ToggleBreadCrumb } from './StyledMainLayout';
import GrowExxTriangleLogo from '../../images/Growexx-Triangle-White.png';
import GrowExxLogo from '../../images/GrowExx_Group_Logo.png';
import { GET_FILTERED_MENU_ITEM } from '../SideBar/constants';
const { Header, Content } = Layout;

const LogoContainer = styled.div`
  background: #190426;
  flex: 0 0 200px;
  max-width: 200px;
  min-width: 200px;
  width: 200px;
  text-align: center;
`;

const HeaderMenu = styled(Menu)`
  display: flex;
  align-items: center;
`;

const HeaderMenuItem = styled(Menu.Item)`
  margin: 0 !important;
  width: fit-content !important;
`;

class Layouts extends React.PureComponent {
  render() {
    const { layoutVariant, collapsed, toggle } = this.props;
    // eslint-disable-next-line default-case
    switch (layoutVariant) {
      case 2:
        return (
          <Layout>
            <Header className="headerLayout">
              <LogoContainer className="logo">
                <Link to={ROUTES.HOME}>
                  {!collapsed ? (
                    <img src={GrowExxLogo} alt="logo" />
                  ) : (
                    <img src={GrowExxTriangleLogo} alt="logo" />
                  )}
                </Link>
              </LogoContainer>
              <AppHeader />
            </Header>
            <Layout className="site-layout">
              <SideBar
                collapsed={collapsed}
                user={getUserData()}
                layoutVariant={layoutVariant}
              />
              <Content
                className="site-layout-background"
                style={{
                  margin: '24px 16px',
                  padding: 24,
                  minHeight: 280,
                }}
              >
                <App />
              </Content>
            </Layout>
            <Footer />
          </Layout>
        );
      case 3:
        return (
          <Layout>
            <Header className="headerLayout">
              <LogoContainer className="logo">
                <Link to={ROUTES.HOME}>
                  {<img src={GrowExxLogo} alt="logo" />}
                </Link>
              </LogoContainer>
              <HeaderMenu
                theme="dark"
                mode="inline"
                defaultSelectedKeys={[this.props.location.pathname]}
                selectedKeys={[this.props.location.pathname]}
              >
                {GET_FILTERED_MENU_ITEM(
                  this.props.user && this.props.user.role,
                ).map(menu => (
                  <HeaderMenuItem key={menu.to} icon={menu.icon}>
                    <Link to={menu.to} />
                  </HeaderMenuItem>
                ))}
              </HeaderMenu>
              <AppHeader menuBackground />
            </Header>
            <Content
              className="site-layout-background"
              style={{
                margin: '24px 16px',
                padding: 24,
                minHeight: 280,
              }}
            >
              <App />
            </Content>
            <Footer />
          </Layout>
        );
      default:
        return (
          <Layout>
            <SideBar
              collapsed={collapsed}
              user={getUserData()}
              layoutVariant={layoutVariant}
            />
            <Layout className="site-layout">
              <Header className="headerLayout">
                <ToggleBreadCrumb>
                  <span
                    className="sideBarTrigger"
                    onClick={toggle}
                    data-testid="ToggleIcon"
                    onKeyDown={toggle}
                    role="button"
                    tabIndex={0}
                    aria-label="Navigation Toggle"
                  >
                    {collapsed ? <MenuUnfoldOutlined /> : <MenuFoldOutlined />}
                  </span>
                </ToggleBreadCrumb>
                <AppHeader />
              </Header>
              <Content
                className="site-layout-background"
                style={{
                  margin: '24px 16px',
                  padding: 24,
                  minHeight: 280,
                }}
              >
                <App />
              </Content>
              <Layout className="site-layout">
                <Footer />
              </Layout>
            </Layout>
          </Layout>
        );
    }
  }
}

Layouts.propTypes = {
  layoutVariant: PropTypes.number,
  collapsed: PropTypes.bool,
  toggle: PropTypes.func,
  location: PropTypes.object,
  user: PropTypes.object,
};

export default withRouter(Layouts);
