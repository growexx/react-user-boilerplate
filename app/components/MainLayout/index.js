/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable react/no-access-state-in-setstate */
/**
 * MainLayout.js
 *
 * This is the entry file for the application, only setup and boilerplate
 * code.
 */

import React from 'react';
import { Layout } from 'antd';
import { MenuUnfoldOutlined, MenuFoldOutlined } from '@ant-design/icons';
import App from 'containers/App';
import Footer from 'components/Footer';
import AppHeader from 'components/Header';
import SideBar from 'components/SideBar';
import { StyledMainLayout, ToggleBreadCrumb } from './StyledMainLayout';
const { Header, Content } = Layout;

class MainLayout extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      collapsed: true,
    };
  }

  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render() {
    return (
      <StyledMainLayout>
        <Layout>
          <SideBar collapsed={this.state.collapsed} />
          <Layout className="site-layout">
            <Header className="headerLayout">
              <ToggleBreadCrumb>
                <span
                  className="sideBarTrigger"
                  onClick={this.toggle}
                  data-testid="ToggleIcon"
                >
                  {this.state.collapsed ? (
                    <MenuUnfoldOutlined />
                  ) : (
                    <MenuFoldOutlined />
                  )}
                </span>
              </ToggleBreadCrumb>
              <AppHeader />
            </Header>
            <Content
              className="site-layout-background"
              style={{
                margin: '24px 16px',
                padding: 24,
                minHeight: 280,
              }}
            >
              <App />
            </Content>
            <Layout className="site-layout">
              <Footer />
            </Layout>
          </Layout>
        </Layout>
      </StyledMainLayout>
    );
  }
}

export default MainLayout;
