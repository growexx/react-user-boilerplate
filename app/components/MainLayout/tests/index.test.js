import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import MainLayout from '../index';
import configureStore from '../../../configureStore';
let store;

const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <MainLayout />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<MainLayout />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should render Div', () => {
    const { container } = componentWrapper();
    const element = container.firstElementChild;
    expect(element.tagName).toEqual('DIV');
  });
  it('should display MenuFoldOutlined icon', () => {
    const { getByTestId } = componentWrapper();
    const element = getByTestId('ToggleIcon');
    fireEvent.click(element);
    expect(element.tagName).toEqual('SPAN');
  });
});
