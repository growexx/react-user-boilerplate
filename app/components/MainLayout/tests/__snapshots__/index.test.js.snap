// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLayout /> should render and match the snapshot 1`] = `
<div
  class="StyledMainLayout-jh4dg-0 jFxBYI"
>
  <section
    class="ant-layout ant-layout-has-sider"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed"
      id="components-layout-demo-custom-trigger"
      style="flex: 0 0 80px; max-width: 80px; min-width: 80px; width: 80px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="logo"
        />
        <ul
          class="ant-menu ant-menu-dark ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
          role="menu"
        >
          <li
            class="ant-menu-item"
            role="menuitem"
          >
            <span
              aria-label="user"
              class="anticon anticon-user ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
            <span>
              <a
                href="/"
              >
                Home
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item ant-menu-item-selected"
            role="menuitem"
          >
            <span
              aria-label="video-camera"
              class="anticon anticon-video-camera ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="video-camera"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM712 792H136V232h576v560zm176-167l-104-59.8V458.9L888 399v226zM208 360h112c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
                />
              </svg>
            </span>
            <span>
              <a
                href="/features"
              >
                Features
              </a>
            </span>
          </li>
        </ul>
      </div>
    </aside>
    <section
      class="ant-layout site-layout"
    >
      <header
        class="ant-layout-header headerLayout"
      >
        <div
          class="StyledMainLayout__ToggleBreadCrumb-jh4dg-1 bxrHpo"
        >
          <span
            class="sideBarTrigger"
            data-testid="ToggleIcon"
          >
            <span
              aria-label="menu-unfold"
              class="anticon anticon-menu-unfold"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="menu-unfold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class="StyledAppHeader-sc-305pn5-0 enPSEw"
        >
          <div
            class="StyledAppHeader__AvatarWrapper-sc-305pn5-1 bMUmVG"
          >
            <span
              aria-label="bell"
              class="anticon anticon-bell"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="bell"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
                />
              </svg>
            </span>
            <div
              class="ant-btn-group ant-dropdown-button"
            >
              <button
                class="ant-btn ant-btn-default"
                type="button"
              />
              <button
                class="ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger"
                type="button"
              >
                <span
                  aria-label="user"
                  class="anticon anticon-user"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content site-layout-background"
        style="margin: 24px 16px; padding: 24px; min-height: 280px;"
      >
        <div
          class="App__AppWrapper-sc-13edzow-0 hLuJAl"
        >
          <div
            class="Wrapper-sc-12uw37d-0 bTWIVF"
          >
            <div
              class="Circle__CirclePrimitive-ww56dy-0 bwXUUe"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 jwPOXH"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 glOezv"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 ihtugw"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 eJJzrd"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 izRsmE"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 cFdckR"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 bnNiyv"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 kkrMUH"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 GRlaQ"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 faWXTY"
            />
            <div
              class="Circle__CirclePrimitive-ww56dy-0 hFmkzH"
            />
          </div>
        </div>
      </main>
      <section
        class="ant-layout site-layout"
      >
        <footer
          class="Wrapper-wcfdfo-0 dPvBog"
        >
          <section>
            <span>
              This project is licensed under the MIT license.
            </span>
          </section>
          <section>
            <div
              class="Wrapper-xnjoq9-0 dNmQIR"
            >
              <select
                class="Select-sc-1sm01tk-0 wIjFa"
              >
                <option
                  value="en"
                >
                  en
                </option>
                <option
                  value="de"
                >
                  de
                </option>
              </select>
            </div>
          </section>
          <section>
            <span>
              
      Made with love by 
              <a
                class="A-br8h0y-0 dwVNvo"
                href="https://twitter.com/mxstbr"
              >
                Max Stoiber
              </a>
              .
    
            </span>
          </section>
        </footer>
      </section>
    </section>
  </section>
</div>
`;
